Orange/widgets/visualize/owscatterplotgraph.py

```diff
@@ -725,8 +729,41 @@ def update_sizes(self):
             size_imputer = getattr(
                 self.master, "impute_sizes", self.default_impute_sizes)
             size_imputer(size_data)
-            self.scatterplot_item.setSize(size_data)
-            self.scatterplot_item_sel.setSize(size_data + SELECTION_WIDTH)
+
+            if self._timer.isActive():
+                self._timer.stop()
+
+            current_size_data = self.scatterplot_item.data["size"]
+            diff = size_data - current_size_data
+
+            class Timeout:
+                factors = [0.2, 0.4, 0.6, 0.8, 1.0]
+
+                def __init__(self, diff_=diff, old_sizes=current_size_data,
+                             item=self.scatterplot_item,
+                             item_sel=self.scatterplot_item_sel):
+                    self._counter = 0
+                    self._diff = diff_
+                    self._old_sizes = old_sizes.copy()
+                    self._scatter_item, self._scatter_item_sel = item, item_sel
+
+                def __call__(self):
+                    factor = self.factors[self._counter]
+                    self._counter += 1
+                    size = self._old_sizes + self._diff * factor
+                    if len(self.factors) == self._counter:
+                        tmr.stop()
```

Comment:
Set the widget's `_timer` to `None`. I believe that otherwise the timer's signal is still connected to an instance of `Timeout()`, which is inside the bound method, which holds a reference to `self`, which contains `_timer` -- and we have circular references and a potential memory leak.